sorc/chgres_cube.fd/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+set(fortran_src
+    atmosphere.F90
+    chgres.F90
+    grib2_util.F90
+    input_data.F90
+    model_grid.F90
+    program_setup.f90
+    search_util.f90
+    static_data.F90
+    surface.F90
+    utils.f90
+    write_data.F90)
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "-g -traceback -r8 -i4 -convert big_endian -assume byterecl")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3 -fp-model precise")
+  set(CMAKE_Fortran_FLAGS_DEBUG
+      "-O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fp-stack-check -fstack-protector-all -fpe0 -debug -ftrapuv"
+  )
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_Fortran_FLAGS "-g -fbacktrace -ffree-line-length-0 -fdefault-real-8 -fconvert=big-endian")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+  set(CMAKE_Fortran_FLAGS_DEBUG
+      "-O0 -ggdb -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check"
+  )
+endif()
```

Comment:
We need to move the default flags one level up for use with every application.
We should only add the application specific ones here to supplement the default flags.

Reply:
Do you mean move CMAKE_Fortran_FLAGS and CMAKE_C_FLAGS to the root level CMakeLists.txt file?

Reply:
I meant all of them
```
CMAKE_Fortran_FLAGS
CMAKE_Fortran_FLAGS_RELEASE
CMAKE_Fortran_FLAGS_DEBUG
```

and the C counterparts.

After that, at this level, we can have
```
set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} --any-application-specifc-flags" )
```
Similarly for RELEASE and DEBUG (depending on the compiler set)
This eliminates repetition. If stock options are good, then the entire section is gone.

Reply:
Done.